makefiles/tools/openocd-adapters/raspi.inc.mk

```diff
@@ -0,0 +1,37 @@
+# Raspberry Pi GPIO as debug adapter
+
+SWCLK ?= 21
+SWDIO ?= 20
+RST ?= 16
+
+ifeq (1, $(shell grep ARMv6 /proc/cpuinfo > /dev/null; echo $?))
+  ifeq (0, $(shell grep [[:lower:]]0311[[:digit:]] /proc/cpuinfo > /dev/null; echo $?))
+    # raspi4
+    PERIPH_BASE  = 0xFE000000
+    SPEED_COEFF  = 236181
+    SPEED_OFFSET = 60
+  else
+    # raspi2, 3
+    PERIPH_BASE  = 0x3F000000
+    SPEED_COEFF  = 146203
+    SPEED_OFFSET = 36
+  endif
+else
+    # raspi1
+    PERIPH_BASE  = 0x20000000
+    SPEED_COEFF  = 113714
+    SPEED_OFFSET = 28
+endif
+
+OPENOCD_ADAPTER_INIT ?= \
+  -c 'interface bcm2835gpio' \
```

Comment:
it doesn't break anything yet but openocd master says this:
> DEPRECATED! use 'adapter driver' not 'interface'

Reply:
I've changed it to use the new ways.

Reply:
I just found that an older version of openocd master from only a few months ago doesn't support the new command yet. So I guess we have to stick with the "deprecated" command. Sorry for that.